### ISSUE TITLE
Collision shape transformation bug

### ISSUE THREAD

**Issue by MrSaberhagen** [godotengine/godot#16083]
**Godot version:**

Godot 3.0 RC2 winx64

**OS/device including version:**

**Issue description:**

I was like coding my first game using tutorial series by KidsCanCode. In third part of the series there was a moment at 7:35 where we were transforming collision shape 2D to fit the players scale (https://www.youtube.com/watch?v=zWAmbgqOWkI). I was doing all exactly the same way, except transforming it not by the right horisontal button, but with the top (vertical) one. And the collision shape just disappeared. Then(without using CTRL + Z) I tried to reshape it with the right button, it appeared back but looked very strange and smashed.

**Steps to reproduce:**
Just make everything as in first three parts of KidsCanCode tutorial and when you get to the moment I mentioned before, try to rescale collision shape with the upper button, not right one.

**Minimal reproduction project:**

**Comment by LikeLakers2** [comment]
Just popping in to note to anyone reading this issue: This tutorial from KidsCanCode is more or less the [Your First Game](http://docs.godotengine.org/en/latest/getting_started/step_by_step/your_first_game.html) tutorial of the Godot docs, so you don't need to follow the video tutorial.

**Comment by LikeLakers2** [comment]
I'm also noticing that the console spams this upon the polygon disappearing:
```
ERROR: Bad Polygon!
   At: servers\visual\visual_server_canvas.cpp:675
```

**Comment by divineforge** [comment]
I can confirm that either. 
I feel odd when using that for the first time, not sure what it's missing suddenly.
It happens to me when I'm trying the **First Game Tutorial** by the Godot Engine Documentation
<img width="390" alt="screen shot 2018-02-03 at 10 04 09 pm" src="https://user-images.githubusercontent.com/1146195/35767877-b2b110ba-092e-11e8-98cb-95a2230b5cd0.png">
<img width="334" alt="screen shot 2018-02-03 at 10 04 13 pm" src="https://user-images.githubusercontent.com/1146195/35767879-b4c8dfe0-092e-11e8-8716-c5cf8a72e5cf.png">

@LikeLakers2  made a good point on the console output. Whenever I adjust the collisionShape close to a 'Circle' shape, it throws the bad polygon message. (Since I'm using a CapsuleShape2D, I assume that's the logic behind, that it can't be a circle)

hence posting my finding above here,

**Comment by marco-vito** [comment]
The issue persist. Anyone know how to work around it?

**Comment by marco-vito** [comment]
I've found the same issue, and bypassed it by changing the roation of the shape. Someone on Discord (thank you, WiggleWizard) sugested there should be an issue trying to set the walls to be smaller than the arcs.

**Comment by raphael10241024** [comment]
Found the reason, when you drag it down, the capsule shape's height will be a negative value, then will cause it can't be painted.

**Comment by epuscasu** [comment]
Same here, once the height goes below 0 the issue occurs.

**Comment by KoBeWi** [comment]
Fixed by #38617
